File: examples/c/primes.c
#include <stdio.h>

void isprime(int *ret, int *arg);
void primes(int *arg, int max, int *ret);


void primes(int *arg, int max, int *ret)
{
    *arg = 2;
    while (*arg < max)
    {
        isprime(ret, arg);
        if (*ret == 1)
            printf("%d\n", *arg);
        *arg = *arg + 1;
    }
}

void isprime(int *ret, int *arg)
{
    int i;

    *ret = 1;
    i = 2;
    while (i < *arg)
    {
        if (*arg / i * i == *arg)
        {
            *ret = 0;
            i = *arg;
        }
        i = i + 1;
    }
}

int main(int argc, char *argv[])
{
    const int max = 100;

    int arg;
    int ret;

    primes(&arg, max, &ret);
    return 0;
}

File: examples/pas/primes.pas
program primes;

    const max = 100;

    var
        arg,
        ret : integer;

    procedure isprime();

        var
            i : integer;

    begin
        ret := 1;
        i := 2;
        while i < arg do
            begin
                if arg div i * i = arg then
                    begin
                        ret := 0;
                        i := arg
                    end;
                i := i + 1
            end
    end;

    procedure primes();

    begin
        arg := 2;
        while arg < max do
            begin
                isprime();
                if ret = 1 then
                    writeln(arg);
                arg := arg + 1
            end
    end;

begin
    primes()
end
.
